homeassistant/components/loqed/sensor.py

```diff
@@ -30,13 +52,41 @@
         entity_category=EntityCategory.DIAGNOSTIC,
         entity_registry_enabled_default=False,
     ),
-    SensorEntityDescription(
+    LoqedSensorEntityDescription(
+        key="wifi_strength",
+        translation_key="wifi_strength",
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    LoqedSensorEntityDescription(
         key="battery_percentage",
         device_class=SensorDeviceClass.BATTERY,
         state_class=SensorStateClass.MEASUREMENT,
         entity_category=EntityCategory.DIAGNOSTIC,
         native_unit_of_measurement=PERCENTAGE,
     ),
+    LoqedSensorEntityDescription(
+        key="battery_voltage",
+        translation_key="battery_voltage",
+        device_class=SensorDeviceClass.VOLTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        entity_registry_enabled_default=False,
+    ),
+    LoqedSensorEntityDescription(
+        key="battery_type",
+        translation_key="battery_type",
+        transform_value=lambda data: str(data).lower(),
```

Comment:
Instead of transforming, it might be better to define a `value_fn` that provides the actual value.

Reply:
I've considered it first too, but it is needed only in two cases.

Reply:
Not really, it would actually improve all cases, as it would get rid of ``getattr` usage (which causes typing to be lost).

Reply:
That makes sense. Will update.

Reply:
Should be fixed.